meilisearch-types/src/keys.rs

```diff
@@ -179,193 +180,256 @@ fn parse_expiration_date(
     }
 }
 
-#[derive(Copy, Clone, Serialize, Deserialize, Debug, Eq, PartialEq, Hash, Sequence, Deserr)]
-#[repr(u8)]
-pub enum Action {
-    #[serde(rename = "*")]
-    #[deserr(rename = "*")]
-    All = 0,
-    #[serde(rename = "search")]
-    #[deserr(rename = "search")]
-    Search,
-    #[serde(rename = "documents.*")]
-    #[deserr(rename = "documents.*")]
-    DocumentsAll,
-    #[serde(rename = "documents.add")]
-    #[deserr(rename = "documents.add")]
-    DocumentsAdd,
-    #[serde(rename = "documents.get")]
-    #[deserr(rename = "documents.get")]
-    DocumentsGet,
-    #[serde(rename = "documents.delete")]
-    #[deserr(rename = "documents.delete")]
-    DocumentsDelete,
-    #[serde(rename = "indexes.*")]
-    #[deserr(rename = "indexes.*")]
-    IndexesAll,
-    #[serde(rename = "indexes.create")]
-    #[deserr(rename = "indexes.create")]
-    IndexesAdd,
-    #[serde(rename = "indexes.get")]
-    #[deserr(rename = "indexes.get")]
-    IndexesGet,
-    #[serde(rename = "indexes.update")]
-    #[deserr(rename = "indexes.update")]
-    IndexesUpdate,
-    #[serde(rename = "indexes.delete")]
-    #[deserr(rename = "indexes.delete")]
-    IndexesDelete,
-    #[serde(rename = "indexes.swap")]
-    #[deserr(rename = "indexes.swap")]
-    IndexesSwap,
-    #[serde(rename = "tasks.*")]
-    #[deserr(rename = "tasks.*")]
-    TasksAll,
-    #[serde(rename = "tasks.cancel")]
-    #[deserr(rename = "tasks.cancel")]
-    TasksCancel,
-    #[serde(rename = "tasks.delete")]
-    #[deserr(rename = "tasks.delete")]
-    TasksDelete,
-    #[serde(rename = "tasks.get")]
-    #[deserr(rename = "tasks.get")]
-    TasksGet,
-    #[serde(rename = "settings.*")]
-    #[deserr(rename = "settings.*")]
-    SettingsAll,
-    #[serde(rename = "settings.get")]
-    #[deserr(rename = "settings.get")]
-    SettingsGet,
-    #[serde(rename = "settings.update")]
-    #[deserr(rename = "settings.update")]
-    SettingsUpdate,
-    #[serde(rename = "stats.*")]
-    #[deserr(rename = "stats.*")]
-    StatsAll,
-    #[serde(rename = "stats.get")]
-    #[deserr(rename = "stats.get")]
-    StatsGet,
-    #[serde(rename = "metrics.*")]
-    #[deserr(rename = "metrics.*")]
-    MetricsAll,
-    #[serde(rename = "metrics.get")]
-    #[deserr(rename = "metrics.get")]
-    MetricsGet,
-    #[serde(rename = "dumps.*")]
-    #[deserr(rename = "dumps.*")]
-    DumpsAll,
-    #[serde(rename = "dumps.create")]
-    #[deserr(rename = "dumps.create")]
-    DumpsCreate,
-    #[serde(rename = "snapshots.*")]
-    #[deserr(rename = "snapshots.*")]
-    SnapshotsAll,
-    #[serde(rename = "snapshots.create")]
-    #[deserr(rename = "snapshots.create")]
-    SnapshotsCreate,
```

Comment:
I got rid of "all" flags that were only referring to a single flag. This caused an issue with implementing `Sequence`, as we couldn't differentiate between an "all" flag and its single referred element, and so we ended up in an infinite loop.